### PR TITLE
KAZOO-4911: don't allow a schedule change to "Scheduled" state withou…

### PR DESCRIPTION
**Pull request by mark2600**
NOTE: This PR should not be merged until the UI changes that are associated have been completed.  They are being tracked in UI-2330.

### CODE REVIEW

**Comment by lazedo** [comment]
@mark2600 please create pr for 3.22

File: applications/crossbar/src/modules/cb_port_requests.erl [hunk 1]
```diff
@@ -381,14 +381,28 @@ patch(Context, Id, ?PORT_SUBMITTED) ->
...
     patch_then_notify(Context, Id, ?PORT_REJECTED);
 patch(Context, Id, ?PORT_CANCELED) ->
     patch_then_notify(Context, Id, ?PORT_CANCELED).
 
+-spec maybe_patch_to_scheduled(cb_context:context(), path_token()) -> cb_context:context().
+maybe_patch_to_scheduled(Context, Id) ->
+    JObj = cb_context:req_data(Context),
+    case kz_json:get_value(<<"scheduled_date">>, JObj) of
+        'undefined' ->
+            cb_context:add_validation_error(<<"error">>, <<"type">>, kz_json:from_list([
```
Comment (fenollp) [inline applications/crossbar/src/modules/cb_port_requests.erl]: Could you drop each argument to add_validation_error to a new line?

Comment (mark2600) [inline applications/crossbar/src/modules/cb_port_requests.erl]: I guess... you people and your love of dropped lists will be the end of me ;)

**Comment by lazedo** [comment]
@mark2600 as discussed, no need for 3.22

**Pull request merged**